pkg/scalers/metrics_api_scaler.go

```diff
@@ -13,20 +13,51 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/metrics/pkg/apis/external_metrics"
 	"net/http"
+	neturl "net/url"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"strconv"
+	"strings"
+	"time"
 )
 
 type metricsAPIScaler struct {
 	metadata *metricsAPIScalerMetadata
+	client   *http.Client
 }
 
 type metricsAPIScalerMetadata struct {
 	targetValue   int
 	url           string
 	valueLocation string
+
+	//apiKeyAuth
+	enableAPIKeyAuth bool
+	method string // +default is header
+	// +optional default header key is X-API-KEY
+	// default query key is api_key
```

Comment:
```suggestion
	method string // way of providing auth key, either "header" using X-API-KEY header or "query" by providing api_key in query params
```
This probably will be more detailed in the docs but I think it's still a good idea to have good clarification in code